### PR TITLE
e2e: Refactor status condition checks to use gomega matchers

### PR DESCRIPTION
**Pull request by timflannagan**
Signed-off-by: timflannagan <timflannagan@gmail.com>

### CODE REVIEW

File: test/e2e/plain_provisioner_test.go [hunk 1]
```diff
@@ -1105,24 +1098,18 @@ var _ = Describe("plain provisioner bundleinstance", func() {
...
-					Reason: rukpakv1alpha1.ReasonInstallationSucceeded,
+					return nil, err
 				}
-				return conditionsSemanticallyEqual(*existing, expected)
-			}).Should(BeTrue())
+				return meta.FindStatusCondition(biOriginal.Status.Conditions, rukpakv1alpha1.TypeInstalled), nil
+			}).Should(And(
+				Not(BeNil()),
+				WithTransform(func(c *metav1.Condition) string { return c.Type }, Equal(rukpakv1alpha1.TypeInstalled)),
+				WithTransform(func(c *metav1.Condition) metav1.ConditionStatus { return c.Status }, Equal(metav1.ConditionFalse)),
```
Comment (joelanford) [inline test/e2e/plain_provisioner_test.go:1109]: To match the original logic?
```suggestion
				WithTransform(func(c *metav1.Condition) metav1.ConditionStatus { return c.Status }, Equal(metav1.ConditionTrue)),
```

**Comment by timflannagan** [comment]
Note: I wasn't entire sure how to refactor https://github.com/operator-framework/rukpak/blob/d8e7a1d1eaa685746e9fd86164253758ac7d26f5/test/e2e/plain_provisioner_test.go#L754 which compares multiple status conditions using a single eventually statement. Open to ideas, but I think we can dissect that check into individual By statements, which allows us to remove that `conditionsSemanticallyEqual` testing helper function.

File: test/e2e/plain_provisioner_test.go (hunk 1 above)
Comment (timflannagan) [reply test/e2e/plain_provisioner_test.go:1109]: Yikes - just saw this in my local e2e results. It looks like there's another e2e check that has a poor copy-paste job that needs some fixing.

**Review by exdx — APPROVED** [review]
Nice, getting rid of those utility functions to use Gomega matchers instead is a big improvement for readability

**Pull request merged**